executor/aggfuncs/func_avg.go

```diff
@@ -102,6 +102,48 @@ func (e *avgOriginal4Decimal) UpdatePartialResult(sctx sessionctx.Context, rowsI
 	return nil
 }
 
+func (e *avgOriginal4Decimal) Slide(sctx sessionctx.Context, rows []chunk.Row, lastStart, lastEnd uint64, shiftStart, shiftEnd uint64, pr PartialResult) error {
+	p := (*partialResult4AvgDecimal)(pr)
+	for i := uint64(0); i < shiftEnd; i++ {
+		input, isNull, err := e.args[0].EvalDecimal(sctx, rows[lastEnd+i])
+		if err != nil {
+			return err
+		}
+		if isNull {
+			continue
+		}
+		if p.count == 0 {
+			p.sum = *input
+			p.count = 1
+			continue
+		}
```

Comment:
Is this path necessary?

Reply:
I think it's really not necessary here.  And there are some similar code in the `aggFuncSum`. Should I create a new pr to remove them?